File: lab02.py
#!/usr/bin/python
#
#sat3310 -lab02
#create by srfranco@mtu.edu
#1/29/2025
#Comments
#
#Variables
#in this section we are just declaring variables and array that we are going to have to use
#Main
import random 
defaultdir = "/home/sat3310/Documents/labs/lab02/data/" 
nouns = "nouns.txt"
agentnouns = "agentnouns.txt"
count = 0
count2 = 0
randomnoun = ""
randomagentnoun = ""
#Open files
with open(defaultdir + nouns, 'rt') as f: #Openinng a file from the default directory named nouns and making it readable and a text file
    with open(defaultdir + agentnouns, 'rt') as f2: #Openinng a file from the default directory named nouns and making it readable and a text file
        for line in f:#Count words with a for loop
            count += 1
        for line in f2:
            count2 += 1
#Put words into a list
mynouns = open(defaultdir + nouns).readlines()#This is making a list called mynouns and going into the default directory and selecting the nouns file copying the informmation from the nouns files into mynouns
myagentnouns = open(defaultdir + agentnouns).readlines()

#Print counts
print ("There are" , count , "words in" , nouns) #using the for loop to count the number of words in the my noun list
print ("There are" , count2 , "words in" , agentnouns) #using the for loop to count the number of words in the my agent noun list

print ("There are", len(mynouns), "words in", nouns) #using the leng of the list to determine the amount of words in mynoun
print ("There are", len(myagentnouns), "words in", agentnouns) #using the leng of the list to determine the amount of words in myagentnoun

#Random select
randomnoun = random.choice(mynouns) # Creating a random noun by selecting from the random noun list
randomagentnoun = random.choice(myagentnouns) # Creating a random agentnoun by selecting from the random agentnoun list

#Print words
print (randomnoun , randomagentnoun) #print the random noun and agent noun

#Done

File: Lab02/lab02.pl
#!/usr/bin/perl
#
#sat3310 -lab02
#create by srfranco@mtu.edu
#1/29/2025
#Comments
#
#Variables
#in this section we are just declaring variables and array that we are going to have to use 
$deaultdir - "/home/sat3310/Documents/labs/lab02/data/";
$nouns = "nouns.txt";
$agentnouns = "agentnouns.txt";
$randomnoun = "";
$randomagentnoun = "";
@arraynouns = [];
@arrayagentnouns = [];
$countnouns = 0;
$countagentnouns = 0;
$DEBUG = "True";
#Main
#
#List Files
#
#Open files
#
open(NOUNSFILE, "<", $defaultdir.$nouns); #opens a file called nounsfile from the vaiable defaultdir adding the file named nouns
open(AGENTNOUNSFILE, "<",$defaultdir.$agentnouns);#opens a file called agentnounsfile from the vaiable defaultdir adding the file named agentnouns

#Close files
@arraynouns = <NOUNSFILE>; #putting the nounsfile into an array
@arrayagentnouns = <AGENTNOUNSFILE>; #putting the agentnouns file into an array

#
#Count words
#
open(NOUNSFILE, "<", $defaultdir.$nouns); #opens a file called nounsfile from the vaiable defaultdir adding the file named nouns
open(AGENTNOUNSFILE, "<",$defaultdir.$agentnouns);#opens a file called agentnounsfile from the vaiable defaultdir adding the file named agentnouns
$countnouns++ while <NOUNSFILE>; #counting the nouns in the file nouns file in a while loop
$countagentnouns++ while <AGENTNOUNSFILE>; #counting the agent nouns in the file agennounsfile

#Put words into an array
#
#@arraynouns = <NOUNSFILE>;
#@arrayagentnouns = <AGENTNOUNSFILE>;

#Print counts
#
print "There are $countnouns words in $nouns\n"; #printing the amount of nouns with a while loop
print "There are $countagentnouns words in $agentnouns\n"; #printing the amount of agent nouns with a while loop
#close NOUNSFILE;
#close AGENTNOUNSFILE;


#Random select
#

$randomnoun = $arraynouns[rand @arraynouns]; # the random noun which is picked from the array of nouns
$randomagentnoun = $arrayagentnouns[rand @arrayagentnouns]; #this is the random noun from the array agent nouns 

#Print Words
#
print "The random noun is $randomnoun\n"; # this is just printing what the random noun is 
print "The random agent noun is $randomagentnoun\n"; #this is just printing what the random agent noun is
print "There are " . scalar @arraynouns . " words in $nouns\n"; #using the size of the file when put into an array with the scalar
print "There are " . scalar @arrayagentnouns . " words in $agentnouns\n"; #using the size of the file when put into an array for agent nouns

#Done!
